### DISCUSSION TITLE
CPU&GPU&TPU

### DISCUSSION THREAD

**Discussion by lx3211** [1101-GS3073-C/Python-Community-GS3073-C#516] [General]
這篇寫得很詳細
還有動圖說明讓人一目瞭然
使用神經網路運算來舉例描述三者的差異

---

**CPU**
- 可以處理文字、控制火箭引擎、執行銀行交易或者使用神經網絡分類圖像
- 神經網絡的大規模運算中的每一步都是完全可預測的，但ALU只能一次一個接著一個地執行它們，每一次都需要訪問內存（馮諾依曼瓶頸）

**GPU**
- 在單個處理器中使用成千上萬個 ALU
- 在有大量並行化的應用中工作得很好
- 但在每次幾千個 ALU 的計算中，GPU 都需要訪問寄存器或共享內存來讀取和保存中間計算結果，也會成比例地耗費更多的能量來訪問內存（馮諾依曼瓶頸）

**TPU**
- 專用於神經網絡工作負載的矩陣處理器
- 不能運行文本處理軟體、控制火箭引擎或執行銀行業務
- 可以為神經網絡處理大量的乘法和加法運算，速度非常快、能耗非常小且物理空間占用也更小
- 主要助因是對馮諾依曼瓶頸的大幅度簡化

---

完整文章：[一次搞懂CPU、GPU 和 TPU](https://kknews.cc/zh-tw/tech/z5qnr4p.html)

**Comment by NcuJerryKao** [comment]
好像課本講義 把三者拿出來做比較 並說明
而例題也可以看到 他們出在一起的選擇體

**Comment by williamyen0623** [comment]
日常生活很少接觸到的TPU原來是拿來做這些事情，長知識了，
感謝分享！

**Comment by BingRong1217** [comment]
透過分享對這些有更多的了解

**Comment by Ben6uo** [comment]
哇哭哇哭 這個東西幫助了我很多

**Comment by HsuChen88** [comment]
我記得TPU一開始是專門為 Google 的TensorFlow框架設計的。

**Comment by HsuChen88** [comment]
後來TPU的發展也相當好，在Google 相簿中，單個 TPU 每天可以處理超過1億張相片。
